### PR TITLE
add jobs to promote e2e testing images

### PR DESCRIPTION
**Pull request by aojea**
add jobs to promote e2e testing images, one job per image.

The job is generated by the scripts in the same folder:
k8s-staging-e2e-test-images.sh

Fixes: https://github.com/kubernetes/kubernetes/issues/96432

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    nonewprivs
+    nonroot
+    pets
+    redis
+    regression-issue-74839
+    resource-consumer
+    sample-apiserver
+    sample-device-plugin
+    volume
+    windows
```
Comment (claudiubelu) [inline config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:40]: In here we have Dockerfiles for a few Windows helper images that are supposed to be built on Windows nodes, something that we don't have in GCB. They aren't meant to be rebuilt, or rebuilt very rarely. You should exclude this.

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    echoserver
+    ipc-utils
+    jessie-dnsutils
+    kitten
+    metadata-concealment
+    nautilus
+    node-perf
+    nonewprivs
+    nonroot
+    pets
```
Comment (claudiubelu) [inline config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:33]: this folder contains a few different images, so it should be something like:

```
pets/peer-finder
pets/redis-installer
pets/zookeeper-installer
```

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    redis
+    regression-issue-74839
+    resource-consumer
+    sample-apiserver
+    sample-device-plugin
+    volume
```
Comment (claudiubelu) [inline config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:39]: ditto

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    apparmor-loader
+    busybox
+    cuda-vector-add
+    echoserver
+    ipc-utils
+    jessie-dnsutils
+    kitten
+    metadata-concealment
+    nautilus
+    node-perf
```
Comment (claudiubelu) [inline config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:30]: same as below.

**Review by claudiubelu — COMMENTED** [review]
I've checked the generator script. Looks good, it generates jobs based on what we already have with the "all-conformance" job and it adds the ``WHAT`` env variable passthrough.

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+cat >"${OUTPUT}" <<EOF
+# Automatically generated by k8s-staging-e2e-test-images.sh.
+
+postsubmits:
+  kubernetes/kubernetes:
+EOF
+
+for image in "${IMAGES[@]}"; do
+    cat >>"${OUTPUT}" <<EOF
+    - name: post-kubernetes-push-e2e-${image//\//-}-test-images
```
Comment (aojea) [inline config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:58]: rhis replaces / by -

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-testing-images
+      decorate: true
+      # we only need to run if the test images have been changed.
+      run_if_changed: '^test\/images\/${image//\//\\/}\/'
```
Comment (aojea) [inline config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:64]: this replace the `/` by `\/` to escape them

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh (hunk 1 above)
Comment (aojea) [reply config/jobs/image-pushing/k8s-staging-e2e-test-images.sh]: done

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh (hunk 3 above)
Comment (aojea) [reply config/jobs/image-pushing/k8s-staging-e2e-test-images.sh]: done

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh (hunk 4 above)
Comment (aojea) [reply config/jobs/image-pushing/k8s-staging-e2e-test-images.sh]: done

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh (hunk 5 above)
Comment (aojea) [reply config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:58]: done

**Comment by aojea** [comment]
kindly ping @spiffxp

File: config/jobs/image-pushing/k8s-staging-e2e-test-images.sh [hunk 7]
```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
```
Comment (BenTheElder) [inline config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:1]: iffy on this file location but I'll allow it I guess :shrug:

Comment (aojea) [reply config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:1]: copied from the original bash file

Comment (spiffxp) [reply config/jobs/image-pushing/k8s-staging-e2e-test-images.sh:1]: I'm not happy with the sludge either but I don't have a better solution off the top of my head.  We could use a pair of verify/update `hack/` scripts to generate jobs using scripts stored in a well-known location?

**Comment by spiffxp** [comment]
/approve
/lgtm

We'll want to sweep through and trigger all of these, and update kubernetes e2e tests to use the new repo location.  However I don't think we should push for that prior to Monday's v1.20 test freeze.

**Comment by spiffxp** [comment]
>We'll want to sweep through and trigger all of these, and update kubernetes e2e tests to use the new repo location. However I don't think we should push for that prior to Monday's v1.20 test freeze.

Opened https://github.com/kubernetes/kubernetes/issues/96770 as followup and put in v1.21 milestone

**Pull request merged**